integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala

```diff
@@ -390,9 +399,9 @@ object CarbonDataRDDFactory {
                 carbonLoadModel,
                 hadoopConf)
             } else if (dataFrame.isDefined) {
-              loadDataFrame(sqlContext, dataFrame, None, carbonLoadModel)
+              loadDataFrame(sqlContext, dataFrame, None, carbonLoadModel, segmentMinMaxAccumulator)
```

Comment:
For compaction, we are recalculating the min max again by reading each block file?
we can just find the min and max from the segments involved in the compaction. No need to compute segment min may by reading each data file.

Reply:
For compaction, we are creating a new load, where we recalculate min and max for writing new index file. I have handled getting segmentmetadata for merged load in compaction flow

Reply:
For compaction, we are creating a new load, where we recalculate min and max for writing new index file. I have handled getting segmentmetadata for merged load in compaction flow